### PR TITLE
Adds the ability to view a node documentation via a node context menu entry. (~2631)

### PR DESCRIPTION
**Pull request by juliusfunke**

### CODE REVIEW

**Comment by juliusfunke** [comment]
Problem: This yields an error when trying to edit a flow until the DocumentationUrl node definition attribute of all nodes in the flow are set. So we can't set the attribute over time as we add the XAML documentation to the node classes, we have to add the attribute to all nodes right away.
I think the best solution is to use the GitHub API the same way we did for the Sybase update last year and add the attribute this way because the URL can be auto-generated in most cases. We would only need to change the URL attribute manually in some cases. Also we could add a basic node documentation template for all nodes when we are at it.

File: src/Simplic.Flow.Editor.UI/ViewModel/Node/NodeViewModel.cs [hunk 1]
```diff
@@ -111,7 +117,54 @@ private void FillPins()
...
+            // Try using all upper case for repo name.
+            url = $"https://simplic.github.io/dev/api_plugins/Simplic%20{repoName.ToUpper()}/api/{fullTypeName}";
+
+            if (CheckURLValid(url))
+                return;
+
+            MessageBox.Show(localizationService.Translate("flow_documentation_not_found"));
+        }
+
+        private bool CheckURLValid(string url)
```
Comment (simplicbe) [inline src/Simplic.Flow.Editor.UI/ViewModel/Node/NodeViewModel.cs:149]: Add description here.

File: src/Simplic.Flow.Editor.UI/ViewModel/Node/NodeViewModel.cs [hunk 2]
```diff
@@ -111,7 +117,54 @@ private void FillPins()
...
+            if (CheckURLValid(url))
+                return;
+
+            // Try using all upper case for repo name.
+            url = $"https://simplic.github.io/dev/api_plugins/Simplic%20{repoName.ToUpper()}/api/{fullTypeName}";
+
+            if (CheckURLValid(url))
+                return;
+
+            MessageBox.Show(localizationService.Translate("flow_documentation_not_found"));
```
Comment (simplicbe) [inline src/Simplic.Flow.Editor.UI/ViewModel/Node/NodeViewModel.cs:146]: All message boxes require:

* Title
* Message
* Button
* Image

**Review by simplicbe — CHANGES_REQUESTED** [review]

File: src/Simplic.Flow.Editor.Definition/Model/NodeDefinition.cs [hunk 3]
```diff
@@ -8,6 +8,15 @@ public abstract class NodeDefinition
         public string DisplayName { get; set; }
         public string Tooltip { get; set; }
         public string Category { get; set; }
+        /// <summary>
+        /// Gets or sets the node documentation URL.
+        /// </summary>
+        public string DocumentationUrl { get; set; }
+        /// <summary>
```
Comment (simplicbe) [inline src/Simplic.Flow.Editor.Definition/Model/NodeDefinition.cs:15]:
```suggestion

        /// <summary>
```

File: src/Simplic.Flow.Editor.UI/Resources.xaml [hunk 4]
```diff
@@ -177,6 +177,13 @@
                 </DataTemplate>
             </Setter.Value>
         </Setter>
+        <Setter Property="ContextMenu">
+            <Setter.Value>
+                <ContextMenu>
+                    <MenuItem Header="Show documentation" Command="{Binding ShowDocumentationCommand}"/>
```
Comment (simplicbe) [inline src/Simplic.Flow.Editor.UI/Resources.xaml:183]: Localization

File: src/Simplic.Flow.Editor.UI/Resources.xaml [hunk 5]
```diff
@@ -191,6 +198,7 @@
         <Setter Property="ContextMenu">
             <Setter.Value>
                 <ContextMenu>
+                    <MenuItem Header="Show documentation" Command="{Binding ShowDocumentationCommand}"/>
```
Comment (simplicbe) [inline src/Simplic.Flow.Editor.UI/Resources.xaml:201]: Localization

**Review by simplicbe — CHANGES_REQUESTED** [review]

File: src/Simplic.Flow/Attribute/NodeDefinitionAttribute.cs [hunk 6]
```diff
@@ -8,5 +8,9 @@ public class NodeDefinitionAttribute : Attribute
         public string DisplayName { get; set; }
         public string Tooltip { get; set; }
         public string Category { get; set; }
+        /// <summary>
```
Comment (simplicbe) [inline src/Simplic.Flow/Attribute/NodeDefinitionAttribute.cs:11]:
```suggestion

        /// <summary>
```

**Pull request merged**